src/plugins/vis_type_timeseries/public/application/components/vis_types/timeseries/vis.js

```diff
@@ -150,10 +169,31 @@ class TimeseriesVisualization extends Component {
     const yAxis = [];
     let mainDomainAdded = false;
 
-    const allSeriesHaveSameFormatters = seriesModel.every(
-      (seriesGroup) => seriesGroup.formatter === seriesModel[0].formatter
+    const allSeriesHaveSameIgnoreFieldFormatting = seriesModel.every(
+      (seriesGroup) =>
```

Comment:
What do you think of creating a utility method for this? At least we will have less code that needs to be converted into typescript + we can write good unit tests for this fragment